components/camera/videosource/logging/logger.go

```diff
@@ -0,0 +1,312 @@
+// Package logging is a thread-safe way to log video device information to a file. On startup, this package creates a
+// unique filename and uses that filename throughout the lifetime of the program to log information such as which video
+// devices are V4L2 compatible and the current operating system.
+package logging
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/pkg/errors"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/config"
+)
+
+var filePath string
+
+func init() {
+	t := time.Now().UTC().Format(time.RFC3339)
+	filePath = filepath.Join(config.ViamDotDir, "debug", "components", "camera", fmt.Sprintf("%s.txt", t))
+}
+
+// InfoMap is a map of information to be written to the log.
+type InfoMap = map[string]string
+
+// Logger is a thread-safe logger that manages a single log file in config.ViamDotDir.
+type Logger struct {
+	infoCh    chan info
+	logger    golog.Logger
+	isRunning atomic.Bool
+	seenPath  map[string]bool
+	seenMap   map[string]InfoMap
+}
+
+type info struct {
+	title string
+	m     InfoMap
+}
+
+const (
+	// keep at most 3 log files in dir.
+	maxFiles = 3
+	linux    = "linux"
+)
+
+// UnsupportedError indicates this feature is not supported on the current platform.
+type UnsupportedError struct{}
+
+func (e UnsupportedError) Error() string {
+	return "unsupported OS"
```

Comment:
[nit] Will it be clear in the logs that the "unsupported OS" error is coming from the camera logger?

Reply:
It is now! Updated.